### PR TITLE
Fully describe validation of render attachments

### LINKED ISSUE #2303: Specification unclear on weather rendering into 3D textures is allowed
https://gpuweb.github.io/gpuweb/#abstract-opdef-gpurenderpasscolorattachment-gpurenderpasscolorattachment-valid-usage and https://gpuweb.github.io/gpuweb/#abstract-opdef-gpurenderpassdepthstencilattachment-gpurenderpassdepthstencilattachment-valid-usage should make clear that views into 3D textures are not allowed.

### PR DESCRIPTION
**Pull request by kainino0x**
Fixes #2303

***
<a href="https://pr-preview.s3.amazonaws.com/kainino0x/gpuweb/pull/2458.html" title="Last updated on Dec 29, 2021, 10:09 PM UTC (c5857f4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/2458/899753b...kainino0x:c5857f4.html" title="Last updated on Dec 29, 2021, 10:09 PM UTC (c5857f4)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -7229,27 +7229,40 @@ dictionary GPURenderPassColorAttachment {
...
+
+    1. Let |renderViewDescriptor| be |this|.{{GPURenderPassColorAttachment/view}}.{{GPUTextureView/[[descriptor]]}}.
+    1. Let |resolveViewDescriptor| be |this|.{{GPURenderPassColorAttachment/resolveTarget}}.{{GPUTextureView/[[texture]]}}.{{GPUTexture/[[descriptor]]}}.
+    1. Let |renderTextureDescriptor| be |this|.{{GPURenderPassColorAttachment/view}}.{{GPUTextureView/[[texture]]}}.{{GPUTexture/[[descriptor]]}}.
+    1. Let |resolveTextureDescriptor| be |this|.{{GPURenderPassColorAttachment/resolveTarget}}.{{GPUTextureView/[[texture]]}}.{{GPUTexture/[[descriptor]]}}.
+
+    The following validation rules apply:
+
+    - |renderViewDescriptor|.{{GPUTextureViewDescriptor/format}} must be a [=color renderable format=].
+    - |this|.{{GPURenderPassColorAttachment/view}} must be a [$renderable texture view$].
```
Comment (kvark) [inline spec/index.bs:7242]: looks like we have a bit of a problem - #2459
that is to say, we can still merge your stuff and follow-up

**Review by kvark — CHANGES_REQUESTED** [review]

File: spec/index.bs [hunk 2]
```diff
@@ -7232,7 +7232,7 @@ dictionary GPURenderPassColorAttachment {
     Given a {{GPURenderPassColorAttachment}} |this|:
 
     1. Let |renderViewDescriptor| be |this|.{{GPURenderPassColorAttachment/view}}.{{GPUTextureView/[[descriptor]]}}.
-    1. Let |resolveViewDescriptor| be |this|.{{GPURenderPassColorAttachment/resolveTarget}}.{{GPUTextureView/[[texture]]}}.{{GPUTexture/[[descriptor]]}}.
+    1. Let |resolveViewDescriptor| be |this|.{{GPURenderPassColorAttachment/resolveTarget}}.{{GPUTexture/[[descriptor]]}}.
```
Comment (kvark) [inline spec/index.bs:7235]: this seems wrong: `resolveTarget` is a view, but it accesses it as `GPUTexture`

**Review by kvark — CHANGES_REQUESTED** [review]

File: spec/index.bs (hunk 2 above)
Comment (kainino0x) [reply spec/index.bs:7235]: oops, nice catch, thanks!

**Review by kvark — APPROVED** [review]
I wish *something* could tell us about these kinds of issues in an automated way

**Pull request merged**